src/core/elevation/qgsabstractprofilegenerator.h

```diff
@@ -198,6 +209,11 @@ class CORE_EXPORT QgsAbstractProfileGenerator
 
     virtual ~QgsAbstractProfileGenerator();
 
+    /**
+     * Returns a unique identifier representing the source of the profile.
+     */
+    virtual QString sourceId() const = 0;
```

Comment:
It looks like we're using layer IDs for the time being. Do we want to comment that those are normally layer IDs, or better to keep the IDs opaque? (i.e. whether anyone could ever rely on converting source ID to layer)

Reply:
Ok, added to the docs. And yes, they'll always match layer IDs for generators associated with a map layer. Other custom generators (e.g. ones from plugins) will have their own interpretation.